### DISCUSSION TITLE
Configure description based upon OS environment

### DISCUSSION THREAD

**Discussion by SeanAlling-DojoFive** [hathach/tinyusb#2625] [Q&A]
In the [net_lwip_webserver](https://github.com/hathach/tinyusb/tree/master/examples/device/net_lwip_webserver) example there is the following in `usb_descriptors.c`

```c++
// Configuration array: RNDIS and CDC-ECM
// - Windows only works with RNDIS
// - MacOS only works with CDC-ECM
// - Linux will work on both
static uint8_t const * const configuration_arr[2] =
{
#if CFG_TUD_ECM_RNDIS
  [CONFIG_ID_RNDIS] = rndis_configuration,
  [CONFIG_ID_ECM  ] = ecm_configuration
#else
  [CONFIG_ID_NCM  ] = ncm_configuration
#endif
```

How can Windows, Linux, or MacOS select the right driver? For example if on Windows. `RNDIS` descriptor should be used whereas for MacOS `CDC-ECM` should be used. 

Testing a device on both MacOS and Windows the correct descriptor is used, but I am unsure how TinyUSB was informed which descriptor to use.

**Comment by hathach** [comment]
tud_descriptor_configuration_cb(index) is called with selected index
